llvm/test/CodeGen/AArch64/sme-intrinsics-stores.ll

```diff
@@ -271,9 +271,9 @@ define void @str_with_off_15(ptr %ptr) {
 define void @str_with_off_15mulvl(ptr %ptr) {
 ; CHECK-LABEL: str_with_off_15mulvl:
 ; CHECK:       // %bb.0:
+; CHECK-NEXT:    incb x0, all, mul #15
```

Comment:
I think this is the same as addvl, right? The optimisation guides state incb is only fast for mul immediates 1,2 and 4. Might be good for now to restrict incb to those cases?

Reply:
It is, though for consistency I thought it would be better to default to INCB, which GCC also [seems to prefer](https://godbolt.org/z/34799Kfqr). Since the slow case of INCB has the same performance as ADDVL, the only problem I saw with this was if we had to read/write from different registers, which ADDVL could do in a single instruction whereas INCB would need a MOV too, but I thought this could be matched from MOV+INCB for example.

That said, I'm happy to restrict INCB to the 1/2/4 forms if you think that's better overall. What do you think?

Reply:
I think it might still be worth being conservative and restricting it to the 1/2/4 forms because for other forms addvl may be better for register allocation, since the input and output registers are not tied.